libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -661,7 +661,9 @@ void WaveClip::WriteXML(XMLWriter &xmlFile) const
 /*! @excsafety{Strong} */
 bool WaveClip::Paste(double t0, const WaveClip& other)
 {
-   if (GetWidth() != other.GetWidth())
+   assert(other.GetPlayDuration() > 0);
```

Comment:
I see in `WaveTrack::ClearAndPaste`, where the code restores the hidden part of the clips, that you now discard `split.left` or `split.right` if empty. I think the result would be the same if you didn't assert here and use the check line 666, right? I'd rather that, because it might keep usage of this method simpler, e.g., you wouldn't need those changes: 
![image](https://github.com/audacity/audacity/assets/22740106/77a8b0b2-7b3a-4921-89cd-2ab979d57624)